[KratosCore][Logging] Adding improved logger

loumalouomega opened the pull request:
@pooyan-dadvand This is the PR after reverting my (WIP) log table

pooyan-dadvand committed:
Adding label to logger message

pooyan-dadvand committed:
Adding label to logger

pooyan-dadvand committed:
corresponding changes to python interface removing the default constructor of logger and logger message

pooyan-dadvand committed:
changing test logger to be compatible with label

pooyan-dadvand committed:
Adding checking category

pooyan-dadvand committed:
Adding KRATOS_CHECK_POINT macro

pooyan-dadvand committed:
Adding test for check point

pooyan-dadvand committed:
Adding KRATOS_INFO macro

roigcarlo committed:
Adding Locator to logger

roigcarlo committed:
Consistent styling

pooyan-dadvand committed:
Making WriteMessage method virtual

roigcarlo committed:
Removing the stack

roigcarlo committed:
Adding Location to the message

roigcarlo committed:
Adding some needed methods

roigcarlo committed:
Adding tests for the new macros

roigcarlo committed:
Merge branch 'core/logger_improvement' of https://github.com/KratosMultiphysics/Kratos into core/logger_improvement

roigcarlo committed:
Making buffers static to avoid segfault

roigcarlo committed:
adding tests for warning

loumalouomega committed:
Merge branch 'master' into core/logger_improvement

loumalouomega committed:
Revert "Restoring python implementation"

This reverts commit 8631fdd2078d7f80c1ac9c257dea5995d79d187d.

loumalouomega committed:
Restoring python implementation

loumalouomega committed:
Restoring python implementation

loumalouomega committed:
WIP logger_table

loumalouomega committed:
Revert "WIP logger_table"

This reverts commit 11dde66f7e1b58bf650d2b91d8d465e9c708c82d.

pooyan-dadvand commented:
I have just realized that I should still add the start and stop functions to Logger. So please do not merge this yet.

philbucher commented on kratos/input_output/logger.h:
```diff
@@ -232,7 +227,32 @@ namespace Kratos
+#define KRATOS_LOG_OCCURRENCES_LINE(line) kratos_log_loop_counter##line
+#define KRATOS_LOG_OCCURRENCES KRATOS_LOG_OCCURRENCES_LINE(__LINE__)
+
+#define KRATOS_INFO(label) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::INFO
```
thumbs up for new macros! 👍

roigcarlo committed:
Changing some logger macros and parameter names

ONCE prints once at the begining of the loop
FIRST is gone
FIRST_N prints the first N times is called

roigcarlo committed:
Merge branch 'core/logger_improvement' of https://github.com/KratosMultiphysics/Kratos into core/logger_improvement

roigcarlo commented:
@pooyan-dadvand Made the changes you told me. Now:
ONCE prints once at the begining of the loop
FIRST is gone
FIRST_N prints the first N times is called
\_\_COUNT\_\_ has been changed to logger_count

pooyan-dadvand commented:
@maceligueta would you please check the effect of logger in performance of the code? you may put some of your outputs as logger and check how it affects the runtime.

maceligueta commented:
@roigcarlo , @pooyan-dadvand , Is it callable from Python? We are printing mostly from the Python side.

pooyan-dadvand commented:
@roigcarlo would you please add the python interface to the wiki?

roigcarlo commented:
@pooyan-dadvand How it works? Didn't remember this having a python interface

pooyan-dadvand commented:
You are right @roigcarlo, the python interface is way behind the cpp version. @maceligueta it takes some time to add the corresponding python interface. We will inform you when we make it work

roigcarlo committed:
Recoding the python interface so its usable

roigcarlo committed:
Changing back to staticmethod impl

roigcarlo committed:
Adding a default Loggeroutput

roigcarlo committed:
Little cleanup

roigcarlo committed:
Adding sverity and scoping context

roigcarlo committed:
Adding severity and scopes

roigcarlo committed:
Moving impl to cpp

pooyan-dadvand committed:
Adding WriteHeader method to LoggerOutput

pooyan-dadvand committed:
Adding LoggerTableOutput class

pooyan-dadvand committed:
Adding test for table logger (Failiing)

pooyan-dadvand committed:
Changing output containers to store a pointer to output

pooyan-dadvand committed:
Adding protected GetStream method to LoggerOutput

pooyan-dadvand committed:
Using the new GetStream method

philbucher commented:
Are you also adding macros for mpi, e.g. to print only on rank zero?

pooyan-dadvand commented:
Of course we do. My idea is to leave the normal macro to write only on rank 0 and have new ones which write for each thread (with rank prefix)

roigcarlo committed:
Adding default print and parsing parameters

pooyan-dadvand committed:
Adding move to column functionality

pooyan-dadvand committed:
Adding more table rows to tests (to be checked)

pooyan-dadvand committed:
Completing the test

loumalouomega commented on kratos/input_output/logger_table_output.cpp:
```diff
@@ -24,7 +24,7 @@
-	LoggerTableOutput::LoggerTableOutput(std::ostream& rOutputStream, std::vector<std::string> const& ColumnsNames) : LoggerOutput(rOutputStream), mColumnsNames(ColumnsNames) {
+	LoggerTableOutput::LoggerTableOutput(std::ostream& rOutputStream, std::vector<std::string> const& ColumnsNames) : LoggerOutput(rOutputStream), mCurrentColumnIndex(0), mColumnsNames(ColumnsNames) {
         for(auto& column_name : mColumnsNames){
             mColumnsWidth.push_back(column_name.size());
```
@pooyan-dadvand Have you rewrite from scratch the table stream?, because in the class table_stream.h there are some tweaks to print correctly in scientific notation and so on

pooyan-dadvand committed:
Ignoring the final spaces and newlines in writing messages

roigcarlo committed:
Merge branch 'master' into core/logger_improvement

roigcarlo committed:
Deleting ERROR because of reasons (hint: visual studio causing problems)

roigcarlo committed:
Some changes to the logger interface

- Logger.print first argument is no longer the label
- Logger.print has the "label" kwarg to be used as a label
- If no label is present in Logger.print, it will default to "undefined"

maceligueta reviewed (approved):
As commented in the meeting at BCN, we should merge this and everyone should test it during their daily coding.

pooyan-dadvand commented:
@loumalouomega there is a test for table output. Take a look at it to see if you can use it.

loumalouomega commented:
Thanks

El 9 feb. 2018 8:59 a. m., "Pooyan Dadvand" <notifications@github.com>
escribió:

> @loumalouomega <https://github.com/loumalouomega> there is a test for
> table output. Take a look at it to see if you can use it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/1406#issuecomment-364362037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEMgGb_EwgrrkaNkV-AbeGzUXi47hHUks5tS_rYgaJpZM4Rycn0>
> .
>
